### ISSUE TITLE
Bad filter processing with 'in' function.

### ISSUE THREAD

**Issue by yaras-phoenix** [robward-scisys/sldeditor#369]
I have many styles containing massive filters with more than 10 literals within 'in' function . For the functions having 2 to 10 literals I can use 'in2', 'in3', ... ,'in10' but when it comes to big ones I need exactly 'in' function so the bug appears. 

- When choosing this style right after the program is opened, I see 'No symbol selected' on the right side of the workspace.
- When choosing this style after viewing another correct style, I see the old style so the right side of the workspace doesn't change.  

And then, when I try to view another correct style again I see the 'Discard changes' box asking me if I want to discard changes (well, there were none) or cancel.

And one more thing: every sld containing 'in' or 'in2'...'in10' functions has blank filter box. When I press 'Edit', the filter dialog says 'Invalid result'. 

Simple example:
[temp.txt](https://github.com/robward-scisys/sldeditor/files/1810784/temp.txt)

**Comment by robward-scisys** [comment]
I've found a couple issues with the SLDEditor but the sld in the temp.txt is actually invalid. The PropertyIsEqualTo first parameter should be a property.  Although the sld is parsed successfully when trying to generate the filter string again, the check is made for the first parameter being a property which throws an exception, hence the filter string is blank.  This is GeoTools functionality and is also present in GeoServer.

I need to find out why the equalTo function is not appearing in the filter dialog and then this will allow the filter string to be correctly configured.

**Comment by yaras-phoenix** [comment]
> The PropertyIsEqualTo first parameter should be a property.

How exactly this should look valid?

<ogc:Filter>
            <ogc:PropertyIsEqualTo>
              <ogc:Function name="in">
               <ogc:PropertyName>TYPE_ID</ogc:PropertyName>
                <ogc:Literal>1</ogc:Literal>
                <ogc:Literal>2</ogc:Literal>
              </ogc:Function>
              <ogc:Literal>true</ogc:Literal>
            </ogc:PropertyIsEqualTo>
           </ogc:Filter>

**Comment by robward-scisys** [comment]
The following appears to be valid:

```
<ogc:Filter>
   <ogc:PropertyIsEqualTo>
       <ogc:Function name="in">
          <ogc:PropertyName>TYPE_ID</ogc:PropertyName>
          <ogc:Literal>1</ogc:Literal>
          <ogc:Literal>2</ogc:Literal>
       </ogc:Function>
       <ogc:Literal>true</ogc:Literal>
   </ogc:PropertyIsEqualTo>
</ogc:Filter>
```
I need to rework the filter panel to allow functions in place of attributes.  I'll work on a fix when I have time.

**Comment by robward-scisys** [comment]
I made some updates, a new jar is available [here](https://www.dropbox.com/sh/w6kwymmdntn2t2l/AAAmlnyqSx3du1uLfJ3WvZ-Ja?dl=0).

I've added the ability to add/remove optional parameters for filter functions that allow it, e.g. the In function.

I've also discovered that the in2, in3, in4 etc functions have been implemented incorrectly up to and including GeoTools version 18.x.  They have been fixed in GeoTools 19 RC1, I haven't updated SLDEditor to use 19 RC1 yet. I need to to a bit of work to make sure SLDEditor will work for Java 8 and Java 9 which I haven't finished yet.

Let me know if there are any issues.
